lib/util/Components.js

```diff
@@ -114,8 +114,12 @@ Components.prototype.list = function() {
     }
     const id = this._getId(this._list[j].node);
     list[j] = this._list[j];
+    if (!list[j].usedPropTypes) {
+      list[j].usedPropTypes = [];
+    }
+
     if (usedPropTypes[id]) {
-      list[j].usedPropTypes = (list[j].usedPropTypes || []).concat(usedPropTypes[id]);
```

Comment:
Yeah, I agree it is not exactly an improvement. 
Reason I added it - so usedPropTypes it is an empty array when there is no usedPropTypes instead of null. Otherwise anywhere we loop through it we would need to check if it is an array or not.
I can remove it and add few checks in the rule file - I don't mind either way. Let me know what is preferable.